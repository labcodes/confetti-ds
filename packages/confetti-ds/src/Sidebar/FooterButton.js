import React from "react";
import PropTypes from "prop-types";
import Icon from "../Icon";
import { ICON_TYPES } from "../constants";

/**
 *
 * @param label This is the button label.
 * @param icon Sets the icon related to the button label. Default state: no icon.
 * @param onClick Callback action to be executed when the FooterButton is clicked.
 * @returns {JSX.Element}
 * @constructor
 */
export default function FooterButton({ label, icon, onClick }) {
  return (
    <button
      type="button"
      className="lab-narrow-sidebar__footer-button"
      onClick={onClick}
    >
      {icon ? (
        <Icon type={icon} className="lab-narrow-sidebar__footer-icon" />
      ) : null}
      <span className="lab-narrow-sidebar__footer-label">{label}</span>
    </button>
  );
}

FooterButton.propTypes = {
  /** This is the button label. */
  label: PropTypes.string.isRequired,
  /** Sets the icon related to the button label. Default state: no icon. */
  icon: PropTypes.oneOf(ICON_TYPES),
  /** Callback action to be executed when the FooterButton is clicked. */
  onClick: PropTypes.func.isRequired,
};

FooterButton.defaultProps = {
  icon: undefined,
};
