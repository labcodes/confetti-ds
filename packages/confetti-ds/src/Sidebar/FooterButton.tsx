import React, { SyntheticEvent } from "react";
import Icon from "../Icon";
import { IconTypes } from "../constants";

interface FooterButtonProps {
  /** This is the button label. */
  label: string,
  /** Sets the icon related to the button label. Default state: no icon. */
  icon?: IconTypes,
  /** Callback action to be executed when the FooterButton is clicked. */
  onClick: (event?: SyntheticEvent) => any,
};

export default function FooterButton({ label, icon, onClick }: FooterButtonProps) {
  return (
    <button
      type="button"
      className="lab-narrow-sidebar__footer-button"
      onClick={onClick}
    >
      {icon ? (
        <Icon type={icon} className="lab-narrow-sidebar__footer-icon" />
      ) : null}
      <span className="lab-narrow-sidebar__footer-label">{label}</span>
    </button>
  );
}
