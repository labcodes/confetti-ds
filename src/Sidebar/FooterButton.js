import React from "react";
import PropTypes from "prop-types";
import Icon from "../Icon";

export default class FooterButton extends React.Component {
  static propTypes = {
    /** This is the button label. */
    label: PropTypes.string.isRequired,
    /** Sets the icon related to the button label. Default state: no icon. */
    icon: PropTypes.string,
    /** Callback action to be executed when the FooterButton is clicked. */
    onClick: PropTypes.func.isRequired,
  };

  static defaultProps = {
    icon: undefined,
  };

  render() {
    const { label, icon, onClick } = this.props;

    return (
      <button
        type="button"
        className="lab-narrow-sidebar__footer-button"
        onClick={onClick}
      >
        {icon ? (
          <Icon type={icon} className="lab-narrow-sidebar__footer-icon" />
        ) : null}
        <span className="lab-narrow-sidebar__footer-label">{label}</span>
      </button>
    );
  }
}
