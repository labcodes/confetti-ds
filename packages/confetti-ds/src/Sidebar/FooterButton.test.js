import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import FooterButton from "./Item";

describe("FooterButton", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(<FooterButton label="Test button" onClick={() => {}} />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with icon", async () => {
    const renderedComponent = renderer
      .create(
        <FooterButton label="Test button" onClick={() => {}} icon="star" />
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("calls props.onClick when clicked", async () => {
    const mockOnClick = jest.fn();
    const shallowButton = shallow(
      <FooterButton label="Test button" onClick={mockOnClick} />
    );
    shallowButton.simulate("click");
    expect(mockOnClick.mock.calls.length).toEqual(1);
  });
});
