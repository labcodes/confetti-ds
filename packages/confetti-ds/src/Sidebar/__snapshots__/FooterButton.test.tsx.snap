// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterButton renders with base props 1`] = `
<button
  className="lab-narrow-sidebar__item"
  onClick={[Function]}
  type="button"
>
  <span
    className="lab-narrow-sidebar__item-label"
  >
    Test button
  </span>
</button>
`;

exports[`FooterButton renders with icon 1`] = `
<button
  className="lab-narrow-sidebar__item"
  onClick={[Function]}
  type="button"
>
  <span
    className="lab-icon lab-icon--star lab-narrow-sidebar__item-icon"
  >
    <svg
      aria-hidden={true}
      className="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
      fill="currentColor"
      height={20}
      viewBox="0 0 20 20"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 2a8 8 0 1 1 0 16 8 8 0 0 1 0-16Zm0 1a7 7 0 1 0 0 14 7 7 0 0 0 0-14Zm0 9.5a.75.75 0 1 1 0 1.5.75.75 0 0 1 0-1.5ZM10 6a.5.5 0 0 1 .5.41V11a.5.5 0 0 1-1 .09V6.5c0-.28.22-.5.5-.5Z"
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    className="lab-narrow-sidebar__item-label"
  >
    Test button
  </span>
</button>
`;
